source/MaterialXCore/Version.cpp

```diff
@@ -1077,6 +1097,107 @@ void Document::upgradeVersion()
                     }
                 }
             }
+            else if (nodeCategory == "swizzle")
+            {
+                InputPtr inInput = node->getInput("in");
+                InputPtr channelsInput = node->getInput("channels");
+                if (inInput &&
```

Comment:
Do you need to handle the case where `inInput` doesn't exist?  All the node defs appear to return `0`, but it's possible the `swizzle` node was wired in to a port that doesn't have `0` as its default?

Reply:
That's a good point, and perhaps we should handle this case by converting the `swizzle` node to a `constant` node containing the appropriate values.  I haven't encountered this case in our test assets, but it's theoretically possible.